src/write_gcm.c

```diff
@@ -1421,8 +1438,16 @@ static int wg_typed_value_create_from_value_t_inline(wg_typed_value_t *result,
         return -1;
       }
       *dataSourceType_static = "gauge";
-      result->field_name_static = "doubleValue";
-      snprintf(buffer, sizeof(buffer), "%f", value.gauge);
+      switch (metric_value_type) {
+        case METRIC_VALUE_TYPE_INT64:
+          result->field_name_static = "int64Value";
+          snprintf(buffer, sizeof(buffer), "%" PRIi64, (int64_t)value.gauge);
+          break;
+        case METRIC_VALUE_TYPE_DOUBLE:
+          result->field_name_static = "doubleValue";
+          snprintf(buffer, sizeof(buffer), "%f", value.gauge);
```

Comment:
`double` requires `%lf`. Also below. Yes, I know it was like this in the original code, but we should fix it.

Reply:
It is not correct to say that double "requires" ```%lf```. 

It's more correct to say that for [fsn]printf, ```%f``` is the correct specifier and ```%lf``` is permitted an an alternative... the "l" is a no-op... it has no effect on %e, %f, %g and friends).

(This is NOT the case for the scanf family, where the exact type is required).

The reason the printf family can be looser like this is that variadic arguments undergo promotion... so all chars become ints, floats become doubles, etc.

Reply:
...and the compiler would have caught it if it were actually a problem, because gcc does format checking these days.

Reply:
Should we update this? WDYT?